src/components/world/SceneInfo.tsx

```diff
@@ -0,0 +1,456 @@
+import React, { MouseEventHandler, MouseEvent } from "react";
+import ReactDOM from "react-dom";
+import { useSelector } from "react-redux";
+import cx from "classnames";
+import debounce from "lodash/debounce";
+import {
+  MAX_ACTORS,
+  MAX_ACTORS_SMALL,
+  MAX_SPRITE_TILES,
+  MAX_TRIGGERS,
+  MAX_ONSCREEN,
+  SCREEN_WIDTH,
+  SCREEN_HEIGHT
+} from "../../consts";
+import { walkSceneEvents } from "lib/helpers/eventSystem";
+import { EVENT_PLAYER_SET_SPRITE } from "lib/compiler/eventTypes";
+import {
+  sceneSelectors,
+  actorSelectors,
+  triggerSelectors,
+  spriteSheetSelectors
+} from "store/features/entities/entitiesState";
+import clamp from "lib/helpers/clamp";
+import l10n from "lib/helpers/l10n";
+import { RootState } from "store/configureStore";
+import { ScriptEvent, SpriteSheet } from "store/features/entities/entitiesTypes";
+
+function useIsMounted() {
+  const isMountedRef = React.useRef(true);
+  React.useEffect(() => {
+    return () => {
+      isMountedRef.current = false;
+    };
+  }, []);
+  return () => isMountedRef.current;
+}
+
+/**
+ * Originally a StackOverflow answer, selected for its use of useRef to avoid
+ * re-rendering and proper debounce assignment, as opposed to re-assigning
+ * the debounce using a useMemo, which would debounce improperly
+ *
+ * @see https://stackoverflow.com/a/62017005/4148154
+ *
+ * One change since then: Strict TS typings
+ */
+function useDebounce<T extends (...args: Array<any>) => any>(cb: T, delay: number) {
+  const inputsRef = React.useRef({ cb, delay });
+  const isMounted = useIsMounted();
+  React.useEffect(() => {
+    inputsRef.current = { cb, delay };
+  });
+  return React.useCallback(
+    debounce((...args: Parameters<T>) => {
+        // Debounce is an async callback. Cancel it, if in the meanwhile
+        // (1) component has been unmounted (see isMounted in snippet)
+        // (2) delay has changed
+        if (inputsRef.current.delay === delay && isMounted())
+          inputsRef.current.cb(...args);
+      },
+      delay
+    ),
+    [delay, debounce]
+  );
+}
```

Comment:
This code _was_ originally pulled from SO but I am confident in its abilities. Pretty familiar with how `useRef` is doing things here, as [I wrote a fairly in-depth article about how `useRef` functions w/ dep arrays](https://unicorn-utterances.com/posts/react-refs-complete-story/)

That said, I'm not _super_ happy with this code living here. Would love to move to a helpers file of some kind. Not familiar enough yet with the codebase structure to decide where that should go. Any thoughts @chrismaltby ?

Reply:
One more reason it would be nice to keep these hooks elsewhere would be because we'll likely be reusing this logic for debouncing the function when we migrate `ColorizedImage`'s `draw` function:

https://github.com/chrismaltby/gb-studio/blob/0e5f99ca4b3935f994ce56d24a5b3613892d37f6/src/components/world/ColorizedImage.js#L35